### PR TITLE
database: remove unused IndexableReposLister.ListPublic

### PR DESCRIPTION
**Pull request by keegancsmith**
This function is unused. So we can simplify logic a bit (onlyPublic is
always false).

Test Plan: mechanical change only affecting sourcegraph.com, so just CI
and code review.

plz-review-url: https://plz.review/review/9963

### CODE REVIEW

File: internal/database/repos.go [hunk 1]
```diff
@@ -1175,8 +1175,10 @@ SELECT repo_id as id FROM user_public_repos WHERE user_id = %d
 `
 
 type ListIndexableReposOptions struct {
-	// If true, will only include uncloned indexable repos
-	OnlyUncloned bool
+	// CloneStatus if set will only return indexable repos of that clone
+	// status.
+	CloneStatus types.CloneStatus
+
 	// If true, we include user added private repos
```
Comment (ggilmore) [inline internal/database/repos.go:1182]: Could you make the documentation style here consistent?:

- make the `IncludePrivate` doc comment consistent with `CloneStatus`' (`IncludePrivate`, if set, will include user-added private repos ...`)
- make the `CloneStatus` doc comment look something like: `If set, CloneStatus will...`

Comment (keegancsmith) [reply internal/database/repos.go:1182]: will do. Out of interest all call sites for this now have IncludePrivate set as true. But the only place this code is used is sourcegraph.com... which only has public repos. So some fun times cleaning this up.

File: cmd/frontend/backend/repos.go [hunk 2]
```diff
@@ -183,8 +183,12 @@ func (s *repos) List(ctx context.Context, opt database.ReposListOptions) (repos
 	return s.store.List(ctx, opt)
 }
 
-// ListIndexable calls database.IndexableRepos.List, with tracing. It lists ALL
-// indexable repos which could include private user added repos.
+// ListIndexable calls database.IndexableRepos.List, with tracing. It lists
+// ALL indexable repos which could include private user added repos. It only
```
Comment (ggilmore) [inline cmd/frontend/backend/repos.go:187]:
```suggestion
// ALL indexable repos which could include private user added repos.
```

File: cmd/frontend/backend/repos.go [hunk 3]
```diff
@@ -183,8 +183,12 @@ func (s *repos) List(ctx context.Context, opt database.ReposListOptions) (repos
...
+// lists cloned repositories.
```
Comment (ggilmore) [inline cmd/frontend/backend/repos.go:188]:
```suggestion
// In addition, it only lists cloned repositories.
```

**Review by ggilmore — APPROVED** [review]

**Review by eseliger — APPROVED** [review]

**Pull request merged**